### ISSUE TITLE
comtypes com object's method returns: 'tuple' object has no attribute '__ctypes_from_outparam__'

### ISSUE THREAD

**Issue by denfromufa** [enthought/comtypes#87]
http://stackoverflow.com/questions/32869070/comtypes-com-objects-method-returns-tuple-object-has-no-attribute-ctypes

I'm invoking a method of COM object with string argument using `comtypes` and method returns:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-102-009507ff0086> in <module>()
----> 1 obj1=xobjData.GetDataType('string_name')

C:\Python\Python27\lib\site-packages\comtypes\__init__.pyc in call_with_inout(self_, *args, **kw)
    657             # be iterable.
    658             if len(outargs) == 1:  # rescode is not iterable
--> 659                 return rescode.__ctypes_from_outparam__()
    660 
    661             rescode = list(rescode)

AttributeError: 'tuple' object has no attribute '__ctypes_from_outparam__'
```

Seems very cryptic error, any help?

`%debug` magic shows following:

```
> c:\python\python27\lib\site-packages\comtypes\__init__.py(659)call_with_inout()
    658             if len(outargs) == 1:  # rescode is not iterable
--> 659                 return rescode.__ctypes_from_outparam__()
    660 

ipdb> outargs
{0: VARIANT(vt=0x8, u'string_name')}
ipdb> rescode
(VARIANT(vt=0x8, u'string_name'), u'Long')
ipdb> exit
```

**Comment by denfromufa** [comment]
some weird possibly related places in code:

```
                    # Note that result may be NULL COM pointer. There is no way
                    # to interpret this properly, so it is returned as-is.

                    # Hm, should we call __ctypes_from_outparam__ on the
                    # result?
                    return result
```

**Comment by denfromufa** [comment]
I patched these places with try, except block:

```
            rescode = func(self_, *args, **kw)
            # If there is only a single output value, then do not expect it to
            # be iterable.
            if len(outargs) == 1:  # rescode is not iterable
                try:
                    return rescode.__ctypes_from_outparam__()
                except: #sometimes rescode is tuple
                    return rescode

            rescode = list(rescode)
            for outnum, o in outargs.items():
                try:
                    rescode[outnum] = o.__ctypes_from_outparam__()
                except:
                    rescode[outnum] = o
            return rescode
```

https://github.com/enthought/comtypes/blob/master/comtypes/__init__.py#L659
https://github.com/enthought/comtypes/blob/master/comtypes/__init__.py#L663

**Comment by denfromufa** [comment]
https://github.com/enthought/comtypes/pull/88

**Comment by junkmd** [comment]
This is resolved by #113.

@denfromufa 
if the issue remains, please re-open.
